src/bufr/BufrParser/Exports/Variables/Transforms/atms/kinds.F90

```diff
@@ -0,0 +1,119 @@
+!            model
+!
+! module history log:
+!   2004-08-15  treadon
+!   2011-07-04  todling - define main precision during compilation
+!
+! Subroutines Included:
+!
+! Functions Included:
+!
+! remarks:
+!   The numerical data types defined in this module are:
+!      i_byte    - specification kind for byte (1-byte) integer variable
+!      i_short   - specification kind for short (2-byte) integer variable
+!      i_long    - specification kind for long (4-byte) integer variable
+!      i_llong   - specification kind for double long (8-byte) integer variable
+!      r_single  - specification kind for single precision (4-byte) real variable
+!      r_double  - specification kind for double precision (8-byte) real variable
+!      r_quad    - specification kind for quad precision (16-byte) real variable
+!
+!      i_kind    - generic specification kind for default integer
+!      r_kind    - generic specification kind for default floating point
+!
+!
+! attributes:
+!   language: f90
+!   machine:  ibm RS/6000 SP
+!
+!$$$ end documentation block
+  implicit none
+  private
+
+! Integer type definitions below
+
+! Integer types
+  integer, parameter, public  :: i_byte  = selected_int_kind(1)      ! byte  integer
+  integer, parameter, public  :: i_short = selected_int_kind(4)      ! short integer
+  integer, parameter, public  :: i_long  = selected_int_kind(8)      ! long  integer
+  integer, parameter, private :: llong_t = selected_int_kind(16)     ! llong integer
+  integer, parameter, public  :: i_llong = max( llong_t, i_long )
+
+! Expected 8-bit byte sizes of the integer kinds
+  integer, parameter, public :: num_bytes_for_i_byte  = 1
+  integer, parameter, public :: num_bytes_for_i_short = 2
+  integer, parameter, public :: num_bytes_for_i_long  = 4
+  integer, parameter, public :: num_bytes_for_i_llong = 8
+
+! Define arrays for default definition
+  integer, parameter, private :: num_i_kinds = 4
+  integer, parameter, dimension( num_i_kinds ), private :: integer_types = (/ &
+       i_byte, i_short, i_long,  i_llong  /) 
+  integer, parameter, dimension( num_i_kinds ), private :: integer_byte_sizes = (/ &
+       num_bytes_for_i_byte, num_bytes_for_i_short, &
+       num_bytes_for_i_long, num_bytes_for_i_llong  /)
+
+! Default values
+! **** CHANGE THE FOLLOWING TO CHANGE THE DEFAULT INTEGER TYPE KIND ***
+  integer, parameter, private :: default_integer = 3  ! 1=byte, 
+                                                      ! 2=short, 
+                                                      ! 3=long, 
+                                                      ! 4=llong
+  integer, parameter, public  :: i_kind = integer_types( default_integer )
+  integer, parameter, public  :: num_bytes_for_i_kind = &
+       integer_byte_sizes( default_integer )
+
+
+! Real definitions below
+
+! Real types
+  integer, parameter, public  :: r_single = selected_real_kind(6)  ! single precision
```

Comment:
This is reinventing the wheel. Just use the iso_fortran_env module then you can use real(real64) or real(real32) etc. Similarly for ints (integer(int32) integer(int64) etc).

Reply:
There are too many of these module floating in jedi. It's not recommended to add one more.

Reply:
@PatNichols  and @emilyhcliu think the `selected_int_kind` ... and other are proliferated, do like the idea of the `iso_fortran_env` as it seems cleaner.